test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -169,3 +177,141 @@
+                   [(date-math op #t "2012-11-21 00:00:00" 2 unit col-type)] [(date-math op #t "2012-11-21 00:00:00" 2 unit col-type)]]
+                  {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                   :fields      [[:expression "expr"]]}]
+
+                 [[[(date-math op #t "2004-03-19 00:00:00" 2 unit col-type)] [(date-math op #t "2008-06-20 00:00:00" 2 unit col-type)]
+                   [(date-math op #t "2012-11-21 00:00:00" 2 unit col-type)] [(date-math op #t "2012-11-21 00:00:00" 2 unit col-type)]]
+                  {:aggregation [[op [:field field-id nil] 2 unit]]}]
+
+                 [(into [] (frequencies
+                             [(date-math op #t "2004-03-19 00:00:00" 2 unit col-type) (date-math op #t "2008-06-20 00:00:00" 2 unit col-type)
+                              (date-math op #t "2012-11-21 00:00:00" 2 unit col-type) (date-math op #t "2012-11-21 00:00:00" 2 unit col-type)]))
+                  {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                   :aggregation [[:count]]
+                   :breakout    [[:expression "expr"]]}]]]
+          (testing (format "%s %s function works as expected on %s column for driver %s" op unit col-type driver/*driver*)
+            (is (= (set expected) (set (test-date-extract query))))))))
+
+   (mt/test-driver :mongo
+     ;; date arithmetic doesn't supports until mongo 5+
+     (when (> (mongo-major-version (mt/db)) 4)
+       (testing "date arithmetic with datetime columns"
+         (let [[col-type field-id] [:datetime (mt/id :times :dt)]]
+           (doseq [op               [:date-add :date-subtract]
+                   unit             [:year :quarter :month :day :hour :minute :second]
+                   [expected query]
+                   [[[[(date-math op #t "2004-03-19 09:19:09" 2 unit col-type)] [(date-math op #t "2008-06-20 10:20:10" 2 unit col-type)]
+                      [(date-math op #t "2012-11-21 11:21:11" 2 unit col-type)] [(date-math op #t "2012-11-21 11:21:11" 2 unit col-type)]]
+                     {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                      :fields      [[:expression "expr"]]}]
+
+                    [(into [] (frequencies
+                                [(date-math op #t "2004-03-19 09:19:09" 2 unit col-type) (date-math op #t "2008-06-20 10:20:10" 2 unit col-type)
+                                 (date-math op #t "2012-11-21 11:21:11" 2 unit col-type) (date-math op #t "2012-11-21 11:21:11" 2 unit col-type)]))
+                     {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                      :aggregation [[:count]]
+                      :breakout    [[:expression "expr"]]}]]]
+             (testing (format "%s %s function works as expected on %s column for driver %s" op unit col-type driver/*driver*)
+               (is (= (set expected) (set (test-date-extract query))))))))
```

Comment:
Mongo-specific tests belong somewhere `modules/drivers/mongo/test`, not in the general tests. Make some of the stuff here public so you can reuse it there if you need to.